enhancements/installer/aws-ingress-nlb-security-group.md

```diff
@@ -0,0 +1,501 @@
+  - More changes upstream increase the consensus/approvals required, especially for new features in service LB on CCM (prefer ALBC in the long term).
+  - More changes in CCM to create and manage the SG lifecycle.
+- CCM/downstream:
+  - More complex to maintain downstream if not fully upstreamed.
+
+Day-2:
+- Self-managed: Patch CIO.
+
+**Phase 2A: Create support on ROSA Classic**
+
+Goals:
+- TBD: Investigate how ROSA Classic (Hive?) reads and reacts to the new option in `install-config`: `platform.aws.ingressController.managedSecurityGroup`. Ensure the CIO manifests are generated correctly.
+
+Day-2:
+- Managed Services: TBD: Patch CIO.
+
+**Phase 2B: Create support on ROSA HCP**
+
+Goals:
+- TBD: Investigate how Hypershift sets the Service Annotation for Security Group management when launching CIO for Hosted Control Planes.
+- TBD: Explore the HCP flow to validate if the self-managed proposed covers it or if specific adjustments are needed.
+
+Day-2:
+- Managed Services: TBD: Patch CIO.
+
+### Workflow Description
+
+#### OpenShift Self-managed
+
+- User:
+  - Creates `install-config.yaml` enabling the use of Security Group, example `platform.aws.ingressController.securityGroupEnabled`, **when** `lbType=NLB` (already exists).
+```yaml
+# install-config.yaml
+platform:
+  aws:
+    region: us-east-1
+    lbType: NLB                   <-- deprecate by platform.aws.ingressController.loadBalancerType?
+    ingressController:            <-- proposing to aggregate CIO configurations
+      securityGroupEnabled: True  <-- new field
+[...]
+```
+- Installer:
+  - The installer will generate the CIO manifests, setting the `managedSecurityGroup` field in the `networkLoadBalancer` section of the IngressController `spec`:
+```yaml
+# $INSTALL_DIR/manifests/cluster-ingress-default-ingresscontroller.yaml
+apiVersion: operator.openshift.io/v1
+kind: IngressController
+metadata:
+  creationTimestamp: null
+  name: default
+  namespace: openshift-ingress-operator
+spec:
+[...]
+  endpointPublishingStrategy:
+    loadBalancer:
+      providerParameters:
+        aws:
+          networkLoadBalancer:
+            managedSecurityGroup: true   <-- new field
+          type: NLB
+        type: AWS
+      scope: External
+    type: LoadBalancerService
+[...]
+```
+- Cluster Ingress Operator (CIO):
+  - CIO will create the Service instance for the default router. If the `spec.endpointPublishingStrategy.loadBalancer.providerParameters.aws.networkLoadBalancer.managedSecurityGroup` is `true`, CIO will include the following annotation in the Service manifest:
+```yaml
+# Manifest for Service XYZ is created with annotations:
+apiVersion: v1
+kind: Service
+metadata:
+  name: echoserver
+  namespace: mrbraga
+  annotations:
+    service.beta.kubernetes.io/aws-load-balancer-type: nlb
+    service.beta.kubernetes.io/aws-load-balancer-scheme: internet-facing
+    service.beta.kubernetes.io/aws-load-balancer-managed-security-group: "true" <-- new annotation
```

Comment:
What does the annotation scheme look like in the AWS LBC? I thought it just allowed you to specify IDs

I think the upstream change to the CCM wants to mimic the behaviour described in https://kubernetes-sigs.github.io/aws-load-balancer-controller/latest/guide/service/annotations/#security-groups

Is our described behaviour here compatible with that, if not, have we deliberately deviated from that pattern?